### ISSUE TITLE
[Bug] after kv int8 quantization, qwen and baichuan chat model return meaningless and repetitive words. 

### ISSUE THREAD

**Issue by zhulinJulia24** [InternLM/lmdeploy#1152]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.

### Describe the bug

after kv int8 quantization, qwen and baichuan chat model return meaningless and repetitive words. It seems qwen doesn't support kv int8 quantization

### Reproduction

```
1. cp -r $path-to-Qwen-14B-Chat /nvme/qa_test_models/Qwen-14B-Chat-inner-kvint8
2. lmdeploy lite calibrate /nvme/qa_test_models/Qwen-14B-Chat --work-dir /nvme/qa_test_models/Qwen-14B-Chat-inner-kvint8
3. lmdeploy chat turbomind /nvme/qa_test_models/Qwen-14B-Chat-inner-kvint8 --tp 2 --model-format hf --quant-policy 4
```

### Environment

```Shell
/home/zhulin1/miniconda3/envs/lmdeployv23/lib/python3.10/site-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning
  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')
sys.platform: linux
Python: 3.10.0 (default, Mar  3 2022, 09:58:08) [GCC 7.5.0]
CUDA available: True
MUSA available: False
numpy_random_seed: 2147483648
GPU 0,1,2,3,4,5,6,7: NVIDIA A100-SXM4-80GB
CUDA_HOME: /usr/local/cuda-11.7
NVCC: Cuda compilation tools, release 11.7, V11.7.64
GCC: gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44)
PyTorch: 2.1.0+cu118
PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201703
  - Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v3.1.1 (Git Hash 64f6bcbcbab628e96f33a62c3e975f8535a7bde4)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - CUDA Runtime 11.8
  - NVCC architecture flags: -gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_90,code=sm_90
  - CuDNN 8.7
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.8, CUDNN_VERSION=8.7.0, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-invalid-partial-specialization -Wno-unused-private-field -Wno-aligned-allocation-unavailable -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.1.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

TorchVision: 0.16.0+cu118
LMDeploy: 0.2.3+
transformers: 4.37.0
gradio: 3.50.2
fastapi: 0.109.2
pydantic: 2.6.1
```

### Error traceback

```Shell
[llama]
model_name = qwen-14b
tensor_para_size = 2
head_num = 40
kv_head_num = 40
vocab_size = 152064
num_layer = 40
inter_size = 13696
norm_eps = 1e-06
attn_bias = 1
start_id = 0
end_id = 151643
session_len = 8200
weight_type = bf16
rotary_embedding = 128
rope_theta = 10000.0
size_per_head = 128
group_size = 0
max_batch_size = 128
max_context_token_num = 1
step_length = 1
cache_max_entry_count = 0.8
cache_block_seq_len = 128
cache_chunk_size = -1
num_tokens_per_iter = 0
max_prefill_iters = 1
extra_tokens_per_iter = 0
use_context_fmha = 1
quant_policy = 4
max_position_embeddings = 2048
rope_scaling_factor = 0.0
use_logn_attn = 1


2024-02-19 13:24:03,807 - lmdeploy - WARNING - get 723 model params
2024-02-19 13:24:14,554 - lmdeploy - WARNING - loading kv_cache quant scale                                                                                                                                                                   
2024-02-19 13:24:14,584 - lmdeploy - WARNING - Input chat template with model_name is None. Forcing to use qwen-14b
[WARNING] gemm_config.in is not found; using default GEMM algo
[WARNING] gemm_config.in is not found; using default GEMM algo
session 1

double enter to end input >>> hello

<|im_start|>system
You are a helpful assistant.<|im_end|>
<|im_start|>user
hello<|im_end|>
<|im_start|>assistant
 2024-02-19 13:24:28,113 - lmdeploy - WARNING - kwargs top_p is deprecated for inference, use GenerationConfig instead.
2024-02-19 13:24:28,113 - lmdeploy - WARNING - kwargs top_k is deprecated for inference, use GenerationConfig instead.
2024-02-19 13:24:28,113 - lmdeploy - WARNING - kwargs temperature is deprecated for inference, use GenerationConfig instead.
2024-02-19 13:24:28,113 - lmdeploy - WARNING - kwargs repetition_penalty is deprecated for inference, use GenerationConfig instead.
2024-02-19 13:24:28,113 - lmdeploy - WARNING - kwargs ignore_eos is deprecated for inference, use GenerationConfig instead.
2024-02-19 13:24:28,113 - lmdeploy - WARNING - kwargs random_seed is deprecated for inference, use GenerationConfig instead.
2024-02-19 13:24:28,113 - lmdeploy - WARNING - kwargs request_output_len is deprecated for inference, use GenerationConfig instead.
HelloHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMAHEMA

double enter to end input >>>
```

**Comment by lvhan028** [comment]
May try our latest main which supports online kv int8 and kv int4.
No more offline calibration is needed.

**Comment by zhulinJulia24** [comment]
fixed in version 0.4.0
